src/Okta.DNX.OAuth.ResourceServer/Controllers/TodoController.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+
+namespace Okta.DNX.OAuth.ResourceServer.Controllers
+{
+    [Authorize("todo.read")]
+    [Route("api/todo")]
+    public class TodoController : Controller
+    {
+        private readonly ILogger _logger;
+
+        public TodoController(Models.ITodoRepository todoItems, ILogger<TodoController> logger)
+        {
+            TodoItems = todoItems;
+            _logger = logger;
+        }
+
+        public Models.ITodoRepository TodoItems { get; set; }
+
+        
+        [HttpGet]
+        public IEnumerable<Models.TodoItem> GetAll()
+        {
+            _logger.LogInformation("Getting all items of Todo Repository");
+            return TodoItems.GetAll();
+        }
+
+        [HttpGet("{id}", Name = "GetTodo")]
+        public IActionResult GetById(string id)
+        {
+            var item = TodoItems.Find(id);
+            if (item == null)
+            {
+                return NotFound();
+            }
+            return new ObjectResult(item);
+        }
+
+        [HttpPost]
+        public IActionResult Create([FromBody] Models.TodoItem item)
+        {
+            if (item == null)
+            {
+                return BadRequest();
+            }
+            TodoItems.Add(item);
+            return CreatedAtRoute("GetTodo", new { id = item.Key }, item);
+        }
+
+        [HttpPut("{id}")]
+        public IActionResult Update(string id, [FromBody] Models.TodoItem item)
```

Comment:
It looks like we're protecting this class with the "todo.read" permissions - is there a different scope that we can use for writes?

I'm okay with either:
1) Changing the name of "todo.read" to something that includes both read/write
2) Adding an extra scope and demonstrating using that for write operations